aspnetcore/signalr/get-started-signalr-core.md

```diff
@@ -0,0 +1,106 @@
+---
+uid: signalr/get-started-signalr-core
+title: Get started with SignalR on ASP.NET Core
+author: rachelappel
+ms.author: rachelap
+description: In this tutorial, you create an app using SignalR for ASP.NET Core.
+manager: wpickett
+ms.date: 02/23/2018
+ms.topic: tutorial
+ms.technology: dotnet-signalr
+ms.prod: aspnet-core
+ms.custom: mvc
+---
+# Tutorial: Get started with SignalR for ASP.NET Core
+
+By [Rachel Appel](https://twitter.com/rachelappel)
+
+This tutorial teaches the basics of building a real-time app using SignalR for ASP.NET Core.
+
+   ![Solution](get-started-signalr-core/_static/signalr-get-started-finished.png)
+
+[View or download the sample](https://github.com/aspnet/Docs/tree/master/aspnetcore/signalr/get-started-signalr-core/sample/). [View download instructions](/tutorials/#how-to-download-a-sample).
+
+This tutorial demonstrates the following SignalR development tasks:
+
+> [!div class="checklist"]
+> * Create an ASP.NET Core web app with a SignalR Hub.
+> * Create a SignalR hub to push content to clients.
+> * Use the SignalR JavaScript library to send messages and display updates from the hub.
+
+# Prerequisites
+
+Install the following software:
+
+* [.NET Core 2.1.0 Preview 1 SDK](https://www.microsoft.com/net/download/dotnet-core/sdk-2.1.300-preview1) or later
+* [Visual Studio 2017](https://www.visualstudio.com/downloads/) version 15.6 or later with the ASP.NET and web development workload
+* [npm](https://www.npmjs.com/get-npm)
```

Comment:
This still doesn't seem accurate to me. I'm almost 100% confident that the ASP.NET workload in VS 2017 15.6 adds npm; however, npm doesn't get added to the path. @jodavis or @justcla, can one of you please confirm?

Reply:
That's correct, we install a sandboxed version of node and npm so we can provide some built-in tools without requiring a full install of Node.js. However, we don't add it to the path generally because it's a somewhat stripped down version and it gets out of date fast. You're better off installing Node.js properly if you want to use it for things.

Installing Node.js should add it to the path if you select the option, and that should make it available in VS, assuming you restarted VS after it was added to the path. If VS was launched from a command prompt, then that also needs to be restarted after Node.js is added to the path.

Reply:
@rachelappel Since we're telling the user to install npm, I think we should also have them modify the **Tools** > **Options** > **Projects and Solutions** > **Web Package Management** > **External Web Tools** setting in VS. Specifically, move `$(PATH)` immediately above `$(VSINSTALLDIR)\Web\External`. This would give their fresh install of npm precedence over what ships with VS.